test/decl/protocol/ownership_protocol.swift

```diff
@@ -0,0 +1,16 @@
+// RUN: %target-typecheck-verify-swift -swift-version 5
+
+class SomeClass {}
+
+protocol P {
+  weak var foo: SomeClass? { get set } // expected-error {{'weak' cannot be applied to a property declaration in a protocol}}
+  unowned var foo2: SomeClass { get set } // expected-error {{'unowned' cannot be applied to a property declaration in a protocol}}
+  weak var foo3: Int? { get set } // expected-error {{'weak' may only be applied to class and class-bound protocol types, not 'Int'}}
+  unowned var foo4: Int { get set } // expected-error {{'unowned' may only be applied to class and class-bound protocol types, not 'Int'}}
+}
+
+extension P {
+  weak var foo5: SomeClass? // expected-error {{extensions may not contain stored properties}}
+  unowned var foo6: SomeClass // expected-error {{extensions may not contain stored properties}}
```

Comment:
Please include some computed properties as well, since we've established that that's currently permitted.

Reply:
Thanks, I've added some tests below of what should still be permitted and not produce an error.